goalie/time_partition.py

```diff
@@ -292,6 +292,19 @@ def __ne__(self, other):
             or not self.field_types == other.field_types
         )
 
+    def drop_last_subinterval(self):
+        """
+        Drop the last subinterval and reset lists and counters appropriately.
+        """
+        self.end_time = self.subintervals[-1][0]
+        self.interval = (self.start_time, self.end_time)
+        self.num_subintervals -= 1
+        self.timesteps.pop(-1)
+        self.subintervals.pop(-1)
+        self.num_timesteps_per_subinterval.pop(-1)
+        self.num_timesteps_per_export.pop(-1)
+        self.num_exports_per_subinterval.pop(-1)
```

Comment:
When adding a new method please implement unit tests.

Reply:
@jwallwork23  see commit https://github.com/mesh-adaptation/goalie/commit/1b358796ff12b49933b2a28ad150d0f570787a37 - added a unit test for the `drop_last_subinterval` method.